pkg/controller/injector/k8s_binding_injector_test.go

```diff
@@ -17,11 +17,175 @@ limitations under the License.
+
+	secretsCl := injector.client.Core().Secrets(binding0.Namespace)
+	secret, err := secretsCl.Get(binding0.Name)
+	if err != nil {
+		t.Fatalf("Error when getting secret: %s", err)
+	}
+	testCredentialsInjected(t, secret.Data, cred0)
+
+	secretsCl = injector.client.Core().Secrets(binding1.Namespace)
+	secret, err = secretsCl.Get(binding1.Name)
+	if err != nil {
+		t.Fatalf("Error when getting secret: %s", err)
+	}
+	testCredentialsInjected(t, secret.Data, cred1)
+}
+
+func TestUninjectOne(t *testing.T) {
+	binding, _ := getBindings()
+	cred, _ := getCreds()
+
+	injector := fakeK8sBindingInjector()
+	inject(t, injector, binding, cred)
+	injector.Uninject(binding)
+
+	secretsCl := injector.client.Core().Secrets(binding.Namespace)
+	secret, err := secretsCl.Get(binding.Name)
+	if err == nil {
+		testCredentialsUninjected(t, secret.Data)
+	}
 }
 
-func TestUninject(t *testing.T) {
+func TestUninjectTwo(t *testing.T) {
+	binding0, binding1 := getBindings()
+	cred0, cred1 := getCreds()
+
+	injector := fakeK8sBindingInjector()
+	inject(t, injector, binding0, cred0)
+	inject(t, injector, binding1, cred1)
+
+	injector.Uninject(binding0)
+
+	// test that binding0 is gone
+	secretsCl := injector.client.Core().Secrets(binding0.Namespace)
+	secret, err := secretsCl.Get(binding0.Name)
+	if err == nil {
+		testCredentialsUninjected(t, secret.Data)
+	}
+
+	//test that binding1 is still there
+	secretsCl = injector.client.Core().Secrets(binding1.Namespace)
+	secret, err = secretsCl.Get(binding1.Name)
+	if err != nil {
+		t.Fatalf("Error when getting secret: %s", err)
+	}
+	testCredentialsInjected(t, secret.Data, cred1)
+
+	// test that binding1 is gone after uninject
+	injector.Uninject(binding1)
+
+	secretsCl = injector.client.Core().Secrets(binding1.Namespace)
+	secret, err = secretsCl.Get(binding1.Name)
+	if err == nil {
+		testCredentialsUninjected(t, secret.Data)
+	}
+}
+
+func fakeK8sBindingInjector() *k8sBindingInjector {
+	return &k8sBindingInjector{
+		client: fake.NewSimpleClientset(),
+	}
+}
+
+func getBindings() (*servicecatalog.Binding, *servicecatalog.Binding) {
+	binding0 := &servicecatalog.Binding{
+		ObjectMeta: kapi.ObjectMeta{
+			Name:      "name0",
+			Namespace: "namespace0",
+		},
+	}
+	binding1 := &servicecatalog.Binding{
+		ObjectMeta: kapi.ObjectMeta{
+			Name:      "name1",
+			Namespace: "namespace1",
+		},
+	}
+	return binding0, binding1
+}
+
+func getCreds() (*brokerapi.Credential, *brokerapi.Credential) {
+	cred0 := &brokerapi.Credential{
+		Hostname: "host0",
+		Port:     "123",
+		Username: "user0",
+		Password: "password!@#!@#!0)",
+	}
+	cred1 := &brokerapi.Credential{
+		Hostname: "host1",
+		Port:     "456",
+		Username: "user1",
+		Password: "password*(&*1)",
+	}
+	return cred0, cred1
+}
+
+func inject(t *testing.T, injector BindingInjector,
+	binding *servicecatalog.Binding, cred *brokerapi.Credential) {
+
+	err := injector.Inject(binding, cred)
+	if err != nil {
+		t.Fatalf("Error when injecting credentials: %s", err)
```

Comment:
this line, and similar ones below, is problematic because the tests above call these functions, but all the failures get reported on this line. that makes it problematic to debug test failures. instead can you return an `error` here and check it in the calling test function?